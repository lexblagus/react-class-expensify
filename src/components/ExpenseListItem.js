import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';


const ExpenseListItem = ({dispatch, id, description, amount, createdAt}) => (
	<div>
		<h2>{description}</h2>
		<button
			onClick = {() => {
				dispatch(removeExpense({id}));
			}}
		>remove</button>
		&nbsp;
		amount = {amount},
		createdAt = {createdAt}
	</div>
);

export default connect()(ExpenseListItem);
