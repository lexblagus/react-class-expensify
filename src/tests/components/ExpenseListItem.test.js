import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem  from '../../components/ExpenseListItem';
import storeDummyData from '../fixtures/storeDummyData';


test('Should render ExpenseListItem with expense', ()=>{
	//console.log( {...storeDummyData[0]} );
	
	const wrapper = shallow(
		<ExpenseListItem
			{...storeDummyData[0]}
			//dispatch    = {()=>{}}
		/>
	);
	
	expect(wrapper).toMatchSnapshot();
});

