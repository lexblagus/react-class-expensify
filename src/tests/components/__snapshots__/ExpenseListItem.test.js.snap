// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpenseListItem with expense 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="columns is-mobile"
    >
      <div
        className="column"
        title={1252015000}
      >
        15-01-1970
      </div>
      <div
        className="column"
      >
        <b>
          1 water bill
        </b>
        <i>
          <Link
            replace={false}
            to="/edit/2da95af6-3181-4902-a9fd-ec52eff5203c"
          />
        </i>
        <br />
        <small>
          (
          <a
            aria-label="remove"
            className="___button ___is-small is-danger"
            href="javascript:;"
            onClick={[Function]}
          >
            delete
          </a>
          )
        </small>
      </div>
      <div
        className="column"
      >
        <b>
          $ 
          100
        </b>
      </div>
    </div>
  </div>
</div>
`;
